src/clj/rems/ext/comanage.clj

```
@@ -47,12 +48,69 @@
+      (if (= 200 (:status response))
+        (delete-org-identity (:OrgIdentityId orglink))
+        (throw (ex-info "Non-200 status code returned: " {:response response}))))
+    (catch Exception e
+      (log/error "Error invoking CoManage DELETE API - " "co_org_identity_links/" (:Id orglink) ".json :" (.getMessage e)))))
+
+(defn get-org-identity-links
+  "Get CoManage organisation identiy links for a user"
+  [copersonid]
+  (try
+    (let [url (str (getx env :comanage-registry-url) "/co_org_identity_links.json?coid=" (getx env :comanage-registry-coid) "&copersonid=" copersonid)
+          response (http/get url (merge +common-opts+ {:basic-auth [(getx env :comanage-core-api-userid) (getx env :comanage-core-api-key)]}))]
+      (if (= 200 (:status response))
+        (let [parsed-json (:body response)
+              orgs (:CoOrgIdentityLinks parsed-json)]
+          orgs)
+        (throw (ex-info "Non-200 status code returned: " {:response response}))))
+    (catch Exception e
+      (log/error "Error invoking CoManage GET API - " "co_org_identity_links.json :" (.getMessage e)))))
+
+(defn get-orcid-identifiers
+  "Get orcid identifiers (if any) from the orgidentityid"
+  [org]
+  (try
+    (let [url (str (getx env :comanage-registry-url) "/identifiers.json?coid=" (getx env :comanage-registry-coid) "&orgidentityid=" (:OrgIdentityId org))
+          response (http/get url (merge +common-opts+ {:basic-auth [(getx env :comanage-core-api-userid) (getx env :comanage-core-api-key)]}))]
+      (if (= 200 (:status response))
+        (let [parsed-json (:body response)
+              identifiers (:Identifiers parsed-json)
+              orcids (filterv #(str/includes? (:Type %) "orcid") identifiers)
+              first-identifier (first orcids)
+              id (if (nil? first-identifier) nil org)]
+          id)
+        (throw (ex-info "Non-200 status code returned: " {:response response}))))
```

Minor, but we can return a better error message if it's a status code of `204`.

[According to the COManage documentation](https://spaces.at.internet2.edu/display/COmanage/Identifier+API), if we get a `204` response it's because the person has no org identity in this case.

Something like: `"The requested Org Identity was found, but has no identifiers attached"`?